kbs/src/api_server.rs

```diff
@@ -129,6 +129,51 @@ impl ApiServer {
             .await
             .map_err(|e| Error::HTTPFailed { source: e.into() })
     }
+
+    /// Setup API server
+    pub fn server(self) -> Result<actix_web::dev::Server> {
```

Comment:
this seems to share a lot of logic with `serve()` in the same file, can we share the code?

Reply:
Ok, I tried a bunch more things and I finally have something that works and avoids the duplication. PTAL